src/Main.hs

```diff
@@ -85,15 +84,15 @@ generateRay :: Camera -> Int -> Int -> Int -> Int -> Ray
 generateRay camera width height x y = 
     let 
          -- Vector from camera to lookAt point
-        eyeVector = buildEyeVector camera 
+        centerVector = eyeVector camera
         -- Vector in local right direction
-        rightVector = normalize (eyeVector `cross` Vector 0 1 0)
-        upVector = normalize (rightVector `cross` eyeVector)
+        rightVector = normalize (centerVector `cross` Vector 0 1 0)
+        upVector = normalize (rightVector `cross` centerVector)
         -- Halves are taken to make right angles
-        halfFov = fovInRadians camera
+        halfFov = fov camera / 2
         -- This aspect ratio will be used, but are not the width and height of the camera
         aspectRatio = h / w
-        halfWidth = tan halfFov
+        halfWidth = tan $ deg2Rad halfFov
```

Comment:
hahahaha, think it's a hangover from all the scala stuff I wrote where the convention for implicit conversions seems to be 2 over to. feel free to change it to non 1337 later